### PR TITLE
Guard against Banner/Interstitial use of nullptr callbacks

### PR DESCRIPTION
**Pull request by DellaBitta**
Check to see if Interstitial and BannerView callback pointers have been nullified before invoking them in Java.

***

Ran iTests on Android phone.
**[iTest Run](https://github.com/firebase/firebase-cpp-sdk/actions/runs/1446187252)**
***

### Type of Change
Place an `x` the applicable box:
- [X] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: admob/src_java/com/google/firebase/admob/internal/cpp/BannerViewHelper.java [hunk 1]
```diff
@@ -177,7 +177,6 @@ public void destroy(final long callbackDataPtr) {
     completeBannerViewFutureCallback(callbackDataPtr,
         ConstantsHelper.CALLBACK_ERROR_NONE,
         ConstantsHelper.CALLBACK_ERROR_MESSAGE_NONE);
-    mNotifyBoundingBoxListenerOnNextDraw.set(true);
```
Comment (DellaBitta) [inline admob/src_java/com/google/firebase/admob/internal/cpp/BannerViewHelper.java:180]: This wouldn't have done anything, anyway, as the mAdView was already set to null.  Also there's a synchronous bounding box changed event immediately prior to this.

**Review by sunmou99 — COMMENTED** [review]
Do we need to add some logs or throw errors when it's a nullptr?

**Comment by DellaBitta** [comment]
> Do we need to add some logs or throw errors when it's a nullptr?

I don't think so.  The most common use case is launching an ad operation and then destroying C++ object before a callback occurs.  

Or more commonly, start the process of destroying the C++ object when suddenly a listener is invoked from Android and iOS about movement, clicks, etc.

**Review by sunmou99 — COMMENTED** [review]
Check no pointer erros. LGTM.

**Review by sunmou99 — APPROVED** [review]

**Pull request merged**